diagnostic_feedback/data_tool.py

```diff
@@ -0,0 +1,161 @@
+import json
+import logging
+from xblock.core import XBlock
+from xblock.fields import Scope, String, Dict, List
+from .sub_api import SubmittingXBlockMixin, my_api
+
+PAGE_SIZE = 15
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+
+log = logging.getLogger(__name__)
+
+
+class ExportDataBlock(XBlock, SubmittingXBlockMixin):
+    active_export_task_id = String(
+        # The UUID of the celery AsyncResult for the most recent export,
+        # IF we are sill waiting for it to finish
+        default="",
+        scope=Scope.user_state,
+    )
+    last_export_result = Dict(
+        # The info dict returned by the most recent successful export.
+        # If the export failed, it will have an "error" key set.
+        default=None,
+        scope=Scope.user_state,
+    )
+    display_data = List(
+        # The list of results associated with the most recent successful export.
+        # Stored separately to avoid the overhead of sending it to the client.
+        default=None,
+        scope=Scope.user_state,
+    )
+
+    def _delete_export(self):
+        self.last_export_result = None
+        self.display_data = None
+        self.active_export_task_id = ''
+
+    def _save_result(self, task_result):
+        """ Given an AsyncResult or EagerResult, save it. """
+        self.active_export_task_id = ''
+        if task_result.successful():
+            if isinstance(task_result.result, dict) and not task_result.result.get('error'):
+                log.info('------------ _save_result- saving result ---------------')
+                self.display_data = task_result.result['display_data']
+                del task_result.result['display_data']
+                self.last_export_result = task_result.result
+            else:
+                self.last_export_result = {'error': u'Unexpected result: {}'.format(repr(task_result.result))}
+                self.display_data = None
+        else:
+            self.last_export_result = {'error': unicode(task_result.result)}
+            self.display_data = None
+
+    @property
+    def download_url_for_last_report(self):
+        """ Get the URL for the last report, if any """
+        # Unfortunately this is a bit inefficient due to the ReportStore API
+        if not self.last_export_result or self.last_export_result['error'] is not None:
+            return None
+        try:
+            from instructor_task.models import ReportStore
+            report_store = ReportStore.from_config(config_name='GRADES_DOWNLOAD')
+            course_key = getattr(self.scope_ids.usage_id, 'course_key', None)
+            return dict(report_store.links_for(course_key)).get(self.last_export_result['report_filename'])
+        except Exception:
+            pass
```

Comment:
Is this to catch an ImportError when instructor_task is not available?  If so, just catch `ImportError`, and then log the error.